### PR TITLE
Feat/add sync networks

### PR DESCRIPTION
**Pull request by asulis**
### Description:
Thais pull request completes the implementation of synchronization service for the directory, adding the implementation for Networks sync 

### Checklist:

_Make sure you tick all the boxes below if they are true or do not apply before you ask for review_

- [ x] I have performed a self-review of my code
- [ x] I have made my code as simple as possible
- [ x] I have added relevant tests for my changes and the code coverage has not dropped substantially
- [ x] I have removed all commented code
- [ x] I have updated the documentation in all relevant places (Javadoc, Swagger, MDs...)
- [ x] I have described the PR and added a meaningful title in the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format

### CODE REVIEW

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericNetwork.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package eu.bbmri_eric.negotiator.dto.discoveryservice;
+
+public interface GenericNetwork {
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericNetwork.java:3]: please add javadoc

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericOrganization.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package eu.bbmri_eric.negotiator.dto.discoveryservice;
+
+public interface GenericOrganization {
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericOrganization.java:3]: missing javadoc

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericResource.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package eu.bbmri_eric.negotiator.dto.discoveryservice;
+
+public interface GenericResource {
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericResource.java:3]: same as above

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/MolgenisCollection.java [hunk 4]
```diff
@@ -7,9 +7,14 @@
 @Getter
 @AllArgsConstructor
 @NoArgsConstructor
-public class MolgenisCollection {
+public class MolgenisCollection implements GenericResource {
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/MolgenisCollection.java:10]: wouldnt it make sense to remove the lombok generated getters and setters for these classes?

File: src/main/java/eu/bbmri_eric/negotiator/service/BBMRIDiscoveryServiceClientImpl.java [hunk 5]
```diff
@@ -235,7 +244,91 @@ private List<MolgenisCollection> findAllCollections() {
     } catch (WebClientResponseException | WebClientRequestException e) {
       log.warn(e.getMessage());
       log.warn("Molgenis is not reachable!");
-      return Collections.emptyList();
+      throw e;
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/service/BBMRIDiscoveryServiceClientImpl.java:247]: shouldnt there be som better handling of this excpetion?

**Review by RadovanTomik — CHANGES_REQUESTED** [review]

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/MolgenisCollection.java (hunk 4 above)
Comment (asulis) [reply src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/MolgenisCollection.java]: Actually they are used when retrieving data from Directory (it is converted into these internal objects). I would leave them for convenience, also considering this is a temporary solution that will disappear once that EMX2 support will be implemented

File: src/main/java/eu/bbmri_eric/negotiator/service/BBMRIDiscoveryServiceClientImpl.java (hunk 5 above)
Comment (asulis) [reply src/main/java/eu/bbmri_eric/negotiator/service/BBMRIDiscoveryServiceClientImpl.java]: There could be but I woudn't waste time on that for the same reasons as before

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericNetwork.java (hunk 1 above)
Comment (asulis) [reply src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericNetwork.java]: Fixed in 3235f439a3ce3690a16b4781597181da46cebe37

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericOrganization.java (hunk 2 above)
Comment (asulis) [reply src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericOrganization.java]: Fixed in 3235f439a3ce3690a16b4781597181da46cebe37

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericResource.java (hunk 3 above)
Comment (asulis) [reply src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/GenericResource.java]: Fixed in 3235f439a3ce3690a16b4781597181da46cebe37

**Review by RadovanTomik — APPROVED** [review]

**Pull request merged**